homeassistant/components/modbus/modbus.py

```diff
@@ -249,17 +251,25 @@ async def async_setup(self):
         for entry in self._pb_call.values():
             entry[ENTRY_FUNC] = getattr(self._client, entry[ENTRY_NAME])
 
+        await self.async_connect_task(None)
+        return True
+
+    async def async_connect_task(self, args):
+        """Try to connect, and retry if needed."""
         async with self._lock:
             if not await self.hass.async_add_executor_job(self._pymodbus_connect):
-                self._log_error("initial connect failed, no retry", error_state=False)
-                return False
+                err = f"{self._config_name} connect failed, retry in {CONNECT_RETRY} seconds"
```

Comment:
Should we mark the entities as unavailable if we fail to connect?

Reply:
That happens already. self._available is False until connected.

Reply:
Looks like it defaults to available until the first update:
https://github.com/home-assistant/core/blob/78f4a49b739b138c51ff46a011c064ab01d24235/homeassistant/components/modbus/base_platform.py#L57

Reply:
Good catch and my bad, I had forgotten that change.

Some entities are in reality not being polled (like a switch), but only updated after a command is sent. These need to have self._available = True, otherwise you cannot issue the command, the rest are now false until the first read (which might be after several update calls, due to the delay: parameter).